_posts/2023-01-24-gpu-jupyterlab-in-galaxy.md

```diff
@@ -0,0 +1,61 @@
+
+
+### Implementation
+
+A Docker container is created that installs several packages such as JupyterLab, TensorFlow, Scikit-learn, Pandas, Bokeh, Elyra AI, Seaborn, ONNX, Git, GPU dashboard, ColabFold, JAX and many others for machine learning and data science projects. The container inherits an official ["NVIDIA/CUDA" base container](https://hub.docker.com/layers/nvidia/cuda/11.8.0-cudnn8-runtime-ubuntu20.04/images/sha256-74b166e2091bb705e9ada685dffe79930612c725669bc87e01125b5245d13f97?context=explore) that contains CUDA packages for GPUs to work with TensorFlow and then installs the above-mentioned packages. The Docker container is downloaded by a [Galaxy interactive tool](https://github.com/usegalaxy-eu/galaxy/blob/release_22.05_europe/tools/interactive/interactivetool_ml_jupyter_notebook.xml) to make it available on Galaxy Europe. Having a Docker container running in the backend provides many security benefits as it interacts minimally with the remote computer's operating system. In addition, a non-root user inside the container provides additional security benefits. These benefits are important as users can execute arbitrary code on JupyterLab notebooks. The Docker container can separately be downloaded to any laptop or personal computer (having at least 20 GBs of space) or any other compute infrastructure from [Docker hub](https://hub.docker.com/layers/anupkumar/docker-ml-jupyterlab/galaxy-integration-0.2/images/sha256-e2d7e28a2f975523db0f5ac29c2e2ce3c7a35b061072098ad388d5b42ee86fba?context=repo) and used. If NVIDIA GPUs are available, the Docker container will automatically recognise them. Otherwise, it will run on CPUs. The scripts to run the container is mentioned in the [GitHub repository](https://github.com/ anuprulez/ml-jupyter-notebook).
```

Comment:
mention that a complete conda environment is setup and a user has access to the entire conda universe

Reply:
The package/plugin that lets user to install conda env does not work well with JupyterLab. It was too buggy, we tried this one: https://pypi.org/project/jupyter-conda/ 

But now, there is also a different one which we can try: https://github.com/mamba-org/gator. Not sure how buggy that would be

Reply:
its not about the plugin, you can still type "!conda install ....". Correct?

Reply:
Yes, that's possible :)